src/reducers/index.js

```diff
@@ -1,12 +1,103 @@
-import advising from './advisingReducer'
-import { combineReducers } from 'redux'
-import courses from './coursesReducer'
-import credits from './creditsReducer'
-import terms from './termsReducer'
-
-export default combineReducers({
-  terms,
-  courses,
-  credits,
-  advising
-})
+const initial_state = {
+  books: '',
+  credits: null,
+  terms: [],
+  courses: [],
+  fetching: false,
+  fetched: false,
+  error: false,
+  term_bounds: null,
+  selected_term: null,
+  events: [],
+}
+
+export default function reducer(state = initial_state, action) {
+  switch (action.type) {
+    case 'FETCH_EVENTS_START':
+    case 'FETCH_COURSES_START':
+    case 'FETCH_SELECTED_COURSES_START': {
+      return { ...state, fetching: true, fetched: false, error: false }
```

Comment:
Interesting - I can't remember, were all of the fetching_start actions always grouped into the same case statement?

It'd seem to me like if a particular fetch failed or we wanted to display a loading icon for just the course content, we wouldn't know which particular fetch we were talking about. I guess if it was always like this then it's something that we haven't needed yet

Reply:
i don't think it's all like this, i just did it this way. i can separate them if you would like, tho

Reply:
i was just trying to avoid adding a bunch of different state variables

Reply:
@amgoodfellow, would you prefer I fix this here or in #212?

Reply:
I think it's fine to go in #212